### PR TITLE
Workaround Worker same origin restriction

### LINKED ISSUE #7566: Worker code must switch to fetch/blob based creation
Creation of web workers from cross origin sources is not allowed unfortunately.

We need to instead

- fetch the source via a CORS request
- create a blob
- launch the Worker with that.

See e.g. this example https://gist.github.com/roman01la/e7a7b879698574c45ca0

This requires the source itself to be CORS accessible, which, coincidentally I need to do anyway because of #7353. See cr/147554802 for that change.

### PR DESCRIPTION
**Pull request by choumx**
Fixes #7566.

/to @cramforce

### CODE REVIEW

File: src/web-worker/amp-worker.js [hunk 1]
```diff
@@ -69,11 +70,25 @@ class AmpWorker {
...
+    /** @private {Worker} */
+    this.worker_ = null;
+
+    /** @const @private {!Promise} */
+    this.fetchPromise_ = this.xhr_.fetchText(url).then(text => {
+      // Fetch web worker binary and create from Blob as workaround since
+      // Worker constructor won't accept CORS URLs.
+      const blob = new win.Blob([text], {type: 'text/javascript'});
+      const blobUrl = win.URL.createObjectURL(blob);
+      this.worker_ = new win.Worker(blobUrl);
```
Comment (cramforce) [inline src/web-worker/amp-worker.js]: Just checked and I'm pretty sure this does not violate our CSP, but we probably should add an explicit `worker-src` CSP directive to lock that down.

File: build-system/server.js [hunk 2]
```diff
@@ -629,7 +629,16 @@ app.get(['/dist/cache-sw.min.html', '/dist/cache-sw.max.html'], function(req, re
  * End Cache SW LOCALDEV section
  */
 
-
+/**
+ * Web worker binary.
+ */
+app.get(['/dist/ww.js', '/dist/ww.max.js'], function(req, res) {
+  fs.readFileAsync(process.cwd() + req.path).then(file => {
+    res.setHeader('Content-Type', 'application/javascript');
```
Comment (jridgewell) [inline build-system/server.js]: Really?

Comment (cramforce) [inline build-system/server.js]: Its what the AMP cache sends :)

File: src/web-worker/amp-worker.js (hunk 1 above)
Comment (choumx) [inline src/web-worker/amp-worker.js]: Good call -- actually it does violate our CSP after this PR due to Blob-ification. 

`worker-src` is CSP3 which has [poor support](https://www.chromestatus.com/features/5922594955984896). Going to add `blob:` to fallback `default-src` instead: cl/147751445

File: src/web-worker/amp-worker.js [hunk 3]
```diff
@@ -69,11 +70,25 @@ class AmpWorker {
...
+    dev().fine(TAG, 'Fetching web worker from:', url);
+
+    /** @private {Worker} */
+    this.worker_ = null;
+
+    /** @const @private {!Promise} */
+    this.fetchPromise_ = this.xhr_.fetchText(url).then(text => {
+      // Fetch web worker binary and create from Blob as workaround since
+      // Worker constructor won't accept CORS URLs.
+      const blob = new win.Blob([text], {type: 'text/javascript'});
```
Comment (choumx) [inline src/web-worker/amp-worker.js]: Looks like `Blob` and `URL.createObjectURL` (Blob URLs) both have [good support](http://caniuse.com/#search=blob).

File: src/web-worker/amp-worker.js (hunk 1 above)
Comment (cramforce) [inline src/web-worker/amp-worker.js]: Please file an issue for Cloudflare to update their CSP. CC @dknecht. 

And update and try out https://github.com/ampproject/amphtml/blob/master/examples/csp.amp.html before submitting cl/147751445

**Comment by choumx** [comment]
@cramforce Friendly ping.

File: src/web-worker/amp-worker.js (hunk 1 above)
Comment (choumx) [inline src/web-worker/amp-worker.js]: - @dknecht confirmed Cloudflare change is on staging
- Verified functionality on `csp.amp.html`

**Pull request merged**